File: front_end/lib/app/core/models/question.dart
import 'answer.dart';

class Question {
  final int id;
  final String content;
  final List<Answer> answers;

  Question({
    this.id = -1,
    required this.content,
    required this.answers,
  });

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'content': content,
      'answers': answers.map((x) => x.toJson()).toList(),
    };
  }

  factory Question.fromJson(Map<String, dynamic> map) {
    return Question(
      id: map['id'] != null
          ? map['id'] as int
          : map['questionId'] != null
              ? map['questionId'] as int
              : -1,
      content: map['content'] as String,
      answers: List<Answer>.from(
        (map['answers'] as List).map<Answer>(
          (x) => Answer.fromJson(x as Map<String, dynamic>),
        ),
      ),
    );
  }
}

File: microservice/result-service/src/main/java/com/example/resultservice/entity/Question.java
package com.example.resultservice.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "question")
public class Question {

    @Id
    @Column(name = "id", length = 10)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "content", length = 255)
    private String content;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "question_id")
    private List<Answer> answers;

    @Column(name = "user_id")
    private int userId;

    public Question() {
    }



    public int getQuestionId() {
        return id;
    }

    public void setQuestionId(int id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public List<Answer> getAnswers(List<Answer> answers) {
        return this.answers;
    }

    public void setAnswers(List<Answer> answers) {
        this.answers = answers;
    }

    public List<Answer> getAnswers() {
        return answers;
    }


    @Override
    public String toString() {
        return "Question{" +
                "id=" + id +
                ", content='" + content + '\'' +
                ", answers=" + answers +
                '}';
    }
}